x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/EvalOperator.java

```diff
@@ -42,12 +42,7 @@ public EvalOperator(BlockFactory blockFactory, ExpressionEvaluator evaluator) {
 
     @Override
     protected Page process(Page page) {
-        Block.Ref ref = evaluator.eval(page);
```

Comment:
question: is it correct to say that, with this PR, `eval` always returns an _owned_ block? I.e. the returned block needs to be closed (is always treated like a floating `Block.Ref` previously was)?

If so, IMO we should update the javadoc for `ExpressionEvaluator.eval` and mention that the returned block must be closed by the caller.

(This is not always the case: `Page.getBlock` for instance returns essentially an un-owned reference to a block and we need to manually `incRef` to take ownership of a page's block.)

Reply:
++ I added some javadocs in https://github.com/elastic/elasticsearch/pull/102513/commits/d56fcf2d005944d29a7bdfd360dcea1a4df51844